File: Algorithms/src/nattybear/level1/SumSigns.java
package nattybear.level1;

class SumSigns {
  public int solution(int[] absolutes, boolean[] signs) {

    int size = signs.length;
    int sum = 0;

    for (int i = 0; i < size; i++) {
      if (signs[i])
        sum += absolutes[i];
      else
        sum += absolutes[i] * -1;
    }

    return sum;
  }
}

File: nattybear/level1/SumSigns.js
function solution(absolutes, signs) {
  const ones = signs.map(x => x ? 1 : -1)
  const numbers = zipWith((a, b) => a * b, absolutes, ones);
  return numbers.reduce((a, b) => a + b, 0);
}

const zipWith = (f, a, b) => a.map((k, i) => f(k, b[i]));
